### PR TITLE
todo30 3

### PR DESCRIPTION
**Pull request by koikonom**
Adress TODO items in preparation for 3.0 beta (13many)

### CODE REVIEW

File: internal/services/compute/orchestrated_virtual_machine_scale_set.go [hunk 1]
```diff
@@ -598,23 +599,24 @@ func OrchestratedVirtualMachineScaleSetDataDiskSchema() *pluginsdk.Schema {
...
-					Optional: true,
-					Computed: true,
-				},
-
-				// TODO 3.0 - change this to ultra_ssd_disk_iops_read_write
-				"disk_mbps_read_write": {
-					Type:     pluginsdk.TypeInt,
-					Optional: true,
-					Computed: true,
-				},
```
Comment (katbyte) [inline internal/services/compute/orchestrated_virtual_machine_scale_set.go:614]: We should probably add the new properties and deprecate the old ones so people have some warning and can move over?

**Review by katbyte — CHANGES_REQUESTED** [review]
just one comment about the renames, we should probably add the news ones and deprecate the old ones (and then put their removable behind the flag) so there sno surprise when they disappear in 3.0

File: internal/services/compute/orchestrated_virtual_machine_scale_set.go [hunk 2]
```diff
@@ -496,22 +497,65 @@ func OrchestratedVirtualMachineScaleSetDataDiskSchema() *pluginsdk.Schema {
...
+
+		o["ultra_ssd_mbps_read_write"] = &pluginsdk.Schema{
+			Type:          pluginsdk.TypeInt,
+			Optional:      true,
+			Computed:      true,
+			ConflictsWith: []string{"disk_mbps_read_write"},
+		}
+	}
+
+	return out
```
Comment (katbyte) [inline internal/services/compute/orchestrated_virtual_machine_scale_set.go:558]: is there a reason we are adding these in 3,0 and not just adding the new properties  deprecating the old ones and removing in 3.0/waiting fort 4.0 for the removable?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/compute/orchestrated_virtual_machine_scale_set.go (hunk 2 above)
Comment (koikonom) [reply internal/services/compute/orchestrated_virtual_machine_scale_set.go]: Alright, based on this and https://github.com/hashicorp/terraform-provider-azurerm/pull/15390#discussion_r808695246 I came up with the approach you can see in the commint below 

(https://github.com/hashicorp/terraform-provider-azurerm/pull/15340/commits/00c3174648a958afe55e7f3ad9f711b67896ae8e)

tl;dr
2.x: There is a deprecation notice to the existing fields
3.x: The fields are renamed.

**Review by katbyte — APPROVED** [review]
LGTM 🚀
